app/models/heart_rate_data.rb

```diff
@@ -0,0 +1,30 @@
+class HeartRateData < ApplicationRecord
+
+  belongs_to :user, dependent: :destroy
+
+  def update_heart_rate_data!
+    update!(data: heart_rate_data_for_the_day)
+  end
+
+  private
+
+  def heart_rate_data_for_the_day
```

Comment:
Do you want `heart_rate_data_for_the_day` to accept a date like `"today"` or are you happy that it will only accept dates in the format `YYYY-mm-dd`?